SparseGrids/tsgCacheLagrange.hpp

```diff
@@ -138,101 +138,58 @@ class CacheLagrangeDerivative : public CacheLagrange<T> {
      *   and the actual nodes with the pre-computed Lagrange coefficients
      * - \b holds the coordinates of the canonical point to cache
      */
-    CacheLagrangeDerivative(int num_dimensions, const std::vector<int> &max_levels, const OneDimensionalWrapper &rule, const double x[]) :
-            CacheLagrange<T>(num_dimensions, max_levels, rule, x) {
-        cacheDerivatives.resize(num_dimensions);
+    CacheLagrangeDerivative(int num_dimensions, const std::vector<int> &max_levels, const OneDimensionalWrapper &rule, const double x[]) {
+        offsets = rule.getPointsCount();
```

Comment:
* initialize the `offsets` and `cache` inline to skip the default initialization, e.g., ` : offsets(rule.getPointsCount())`

Reply:
Changed.